src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java

```diff
@@ -315,16 +316,21 @@ public WorkflowRun(WorkflowJob job, File dir) throws IOException {
                 executionLoaded = true;
                 execution = newExecution;
             }
-
-            newExecution.start();
-            executionPromise.set(newExecution);
-            FlowExecutionListener.fireRunning(execution);
+            SettableFuture<FlowExecution> exec = getSettableExecutionPromise();
+            if (!exec.isDone()) {
+                exec.set(newExecution);
```

Comment:
Minor race condition I suppose, though presumably in `run` of a new build there could be no confusion about which value is correct.